### PR TITLE
Add Textarea and CheckboxList components

### PR DESCRIPTION
**Pull request by halkyon**

### CODE REVIEW

File: js/form.jsx [hunk 1]
```diff
@@ -406,6 +434,105 @@ var Select = React.createClass({
...
+				value: value,
+				title: this.props.values[value],
+				selected: _.contains(this.props.value, value)
+			}
+		}.bind(this));
+
+		return {
+			isValid: true,
+			data: data,
+			value: this.props.value || [],
```
Comment (mateusz) [inline js/form.jsx]: Can we rename either props.values or props.value just to distinguish them?

File: js/form.jsx [hunk 2]
```diff
@@ -406,6 +434,105 @@ var Select = React.createClass({
...
+		}
+	},
+
+	componentWillUnmount: function() {
+		this.props.detachFromForm(this);
+	},
+
+	setValue: function(value) {
+		var values = [];
+		var state = this.state.data.map(function(item) {
```
Comment (mateusz) [inline js/form.jsx]: state => newData?

File: js/form.jsx [hunk 3]
```diff
@@ -406,6 +434,105 @@ var Select = React.createClass({
...
+		if (this.state.value!=nextState.value) {
+			if (_.has(this.props, 'onSetValue')) this.props.onSetValue(nextState.value);
+		}
+	},
+
+	componentWillUnmount: function() {
+		this.props.detachFromForm(this);
+	},
+
+	setValue: function(value) {
```
Comment (mateusz) [inline js/form.jsx]: setValue => handleCheckboxChange

File: js/form.jsx (hunk 1 above)
Comment (halkyon) [inline js/form.jsx]: Done! values -> source, value remained intact.

**Comment by mateusz** [comment]
:pipboy:

**Pull request merged**